src/exp/kafka-watcher/cmd/main.go

```diff
@@ -47,6 +56,10 @@ func main() {
 		logrus.WithError(err).Panic()
 	}
 
+	if err = w.ValidateKafkaServers(signals.SetupSignalHandler()); err != nil {
```

Comment:
I think you can't call signals.SetupSignalHandler twice - set it up once and then pass the context to both